diff -Naur subliminal-0.7.5/subliminal/cli.py subliminal-0.7.5.patched/subliminal/cli.py
--- subliminal-0.7.5/subliminal/cli.py	2015-03-04 17:42:26.000000000 -0500
+++ subliminal-0.7.5.patched/subliminal/cli.py	2016-05-01 16:04:46.917796000 -0400
@@ -52,6 +52,13 @@
     filtering_group.add_argument('-f', '--force', action='store_true',
                                  help='force subtitle download for videos with existing subtitles')
 
+    # addic7ed
+    addic7ed_group = parser.add_argument_group('addic7ed')
+    addic7ed_group.add_argument('--addic7ed-username', metavar='USERNAME',
+                                help='username for addic7ed provider')
+    addic7ed_group.add_argument('--addic7ed-password', metavar='PASSWORD',
+                                help='password for addic7ed provider')
+
     # output
     output_group = parser.add_argument_group('output')
     output_exclusive_group = output_group.add_mutually_exclusive_group()
@@ -95,6 +102,10 @@
 
     # parse provider configs
     provider_configs = {}
+    provider_configs['addic7ed'] = {
+        'username': args.addic7ed_username,
+        'password': args.addic7ed_password,
+    }
 
     # parse color
     if args.color and colorlog is None:
diff -Naur subliminal-0.7.5/subliminal/providers/addic7ed.py subliminal-0.7.5.patched/subliminal/providers/addic7ed.py
--- subliminal-0.7.5/subliminal/providers/addic7ed.py	2016-02-05 08:46:15.660263000 -0500
+++ subliminal-0.7.5.patched/subliminal/providers/addic7ed.py	2016-05-01 16:19:41.684206000 -0400
@@ -51,6 +51,10 @@
 
 
 class Addic7edProvider(Provider):
+    # Defaults
+    username = None
+    password = None
+
     languages = set([babelfish.Language('por', 'BR')]) | set([babelfish.Language(l)
                  for l in ['ara', 'aze', 'ben', 'bos', 'bul', 'cat', 'ces', 'dan', 'deu', 'ell', 'eng', 'eus', 'fas',
                            'fin', 'fra', 'glg', 'heb', 'hrv', 'hun', 'hye', 'ind', 'ita', 'jpn', 'kor', 'mkd', 'msa',
@@ -59,12 +63,71 @@
     video_types = (Episode,)
     server = 'http://www.addic7ed.com'
 
+    def __init__(self, username=None, password=None):
+
+        self.logged_in = None
+        if username and password:
+            logger.info('Addic7ed using authentication serice.')
+            self.username = username
+            self.password = password
+            self.logged_in = False
+        else:
+            logger.info('Addic7ed using non-authenticated service.')
+
+
     def initialize(self):
         self.session = requests.Session()
         self.session.headers = {
             'User-Agent': self.random_user_agent,
             'Referer': self.server,
         }
+        if self.logged_in is False:
+            # Attempt to log in
+            logger.debug('Preparing to log into Addic7ed...')
+            data = {
+                'username': self.username,
+                'password': self.password,
+                'Submit': 'Log in',
+            }
+            try:
+                r = self.session.post(
+                    self.server + '/dologin.php', data, timeout=10,
+                    allow_redirects=False,
+                )
+            except requests.Timeout:
+                raise ProviderNotAvailable(
+                    'Addic7ed Authentication timeout',
+                )
+
+            if r.status_code == 302:
+                logger.debug('Successfully authenticated with Addic7ed.')
+                self.logged_in = True
+            else:
+                logger.warning('Failed to authenticate with Addic7ed!')
+
+    def terminate(self):
+        # logout
+        if self.logged_in:
+
+            # Toggle our flag reguardless of our success
+            self.logged_in = False
+            try:
+                r = self.session.get(self.server + '/logout.php', timeout=10)
+                logger.debug('Successfully logged out of Addic7ed.')
+            except requests.Timeout:
+                # No problem... we're done anyway
+                logger.warning('A timeout occured logging out of Addic7ed!')
+                return
+
+            if r.status_code != 200:
+                logger.warning(
+                    'Addic7ed returned the error code %d while logging out' %\
+                    r.status_code,
+                )
+                return
+
+        # Close our session
+        self.session.close()
 
     def get(self, url, params=None):
         """Make a GET request on `url` with the given parameters
