diff -Naur subliminal-0.7.4/subliminal/providers/podnapisi.py subliminal-0.7.4.patched/subliminal/providers/podnapisi.py
--- subliminal-0.7.4/subliminal/providers/podnapisi.py	2014-01-27 16:17:14.000000000 -0500
+++ subliminal-0.7.4.patched/subliminal/providers/podnapisi.py	2014-07-13 18:55:04.000000000 -0400
@@ -121,15 +121,25 @@
                 logger.debug('No subtitle found')
                 break
             if series and season and episode:
-                subtitles.extend([PodnapisiSubtitle(language, int(s.find('id').text), s.find('release').text.split(),
+                try:
+                    subtitles.extend([PodnapisiSubtitle(language, int(s.find('id').text), s.find('release').text.split(),
                                                     'h' in (s.find('flags').text or ''), s.find('url').text[38:],
                                                     series=series, season=season, episode=episode)
                                   for s in root.findall('subtitle')])
+                except AttributeError:
+                    # there simply wasn't enough information in the TV Show
+                    # gracefully handle this instead of crashing :)
+                    break
             elif title:
-                subtitles.extend([PodnapisiSubtitle(language, int(s.find('id').text), s.find('release').text.split(),
+                try:
+                    subtitles.extend([PodnapisiSubtitle(language, int(s.find('id').text), s.find('release').text.split(),
                                                     'h' in (s.find('flags').text or ''), s.find('url').text[38:],
                                                     title=title, year=year)
                                   for s in root.findall('subtitle')])
+                except AttributeError:
+                    # there simply wasn't enough information in the movie
+                    # gracefully handle this instead of crashing :)
+                    break
             if int(root.find('pagination/current').text) >= int(root.find('pagination/count').text):
                 break
             params['page'] = int(root.find('pagination/current').text) + 1
