diff -Naur subliminal-0.7.4/subliminal/cache.py subliminal-0.7.4.patched/subliminal/cache.py
--- subliminal-0.7.4/subliminal/cache.py	2014-01-27 16:17:14.000000000 -0500
+++ subliminal-0.7.4.patched/subliminal/cache.py	2014-10-18 16:08:52.000000000 -0400
@@ -7,7 +7,7 @@
 
 
 #: Subliminal's cache version
-CACHE_VERSION = 1
+CACHE_VERSION = 2
 
 
 def subliminal_key_generator(namespace, fn, to_str=string_type):
diff -Naur subliminal-0.7.4/subliminal/providers/addic7ed.py subliminal-0.7.4.patched/subliminal/providers/addic7ed.py
--- subliminal-0.7.4/subliminal/providers/addic7ed.py	2014-10-18 16:02:47.000000000 -0400
+++ subliminal-0.7.4.patched/subliminal/providers/addic7ed.py	2014-10-18 16:05:32.000000000 -0400
@@ -6,6 +6,7 @@
 import charade
 import requests
 from . import Provider
+from . import IGNORED_CHARACTERS_RE
 from .. import __version__
 from ..cache import region
 from ..exceptions import ProviderConfigurationError, ProviderNotAvailable, InvalidSubtitle
@@ -126,7 +127,9 @@
         soup = self.get('/shows.php')
         show_ids = {}
         for html_show in soup.select('td.version > h3 > a[href^="/show/"]'):
-            show_ids[html_show.string.lower()] = int(html_show['href'][6:])
+            show_ids[
+                IGNORED_CHARACTERS_RE.sub('', html_show.string).lower()] = \
+                    int(html_show['href'][6:])
         return show_ids
 
     @region.cache_on_arguments()
@@ -150,10 +153,11 @@
 
     def query(self, series, season):
         show_ids = self.get_show_ids()
-        if series.lower() in show_ids:
-            show_id = show_ids[series.lower()]
+        _series = IGNORED_CHARACTERS_RE.sub('', series).lower()
+        if _series in show_ids:
+            show_id = show_ids[_series]
         else:
-            show_id = self.find_show_id(series.lower())
+            show_id = self.find_show_id(_series)
             if show_id is None:
                 return []
         params = {'show_id': show_id, 'season': season}
diff -Naur subliminal-0.7.4/subliminal/providers/__init__.py subliminal-0.7.4.patched/subliminal/providers/__init__.py
--- subliminal-0.7.4/subliminal/providers/__init__.py	2014-01-27 16:17:14.000000000 -0500
+++ subliminal-0.7.4.patched/subliminal/providers/__init__.py	2014-10-18 16:04:58.000000000 -0400
@@ -3,6 +3,9 @@
 import babelfish
 from ..video import Episode, Movie
 
+import re
+#:  The following characters are always stripped
+IGNORED_CHARACTERS_RE = re.compile('[!@#$\'"]')
 
 class Provider(object):
     """Base class for providers
diff -Naur subliminal-0.7.4/subliminal/providers/opensubtitles.py subliminal-0.7.4.patched/subliminal/providers/opensubtitles.py
--- subliminal-0.7.4/subliminal/providers/opensubtitles.py	2014-10-18 16:02:47.000000000 -0400
+++ subliminal-0.7.4.patched/subliminal/providers/opensubtitles.py	2014-10-18 16:05:52.000000000 -0400
@@ -10,6 +10,7 @@
 import charade
 import guessit
 from . import Provider
+from . import IGNORED_CHARACTERS_RE
 from .. import __version__
 from ..exceptions import ProviderError, ProviderNotAvailable, InvalidSubtitle
 from ..subtitle import Subtitle, is_valid_subtitle, compute_guess_matches
@@ -50,7 +51,9 @@
         # episode
         if isinstance(video, Episode) and self.movie_kind == 'episode':
             # series
-            if video.series and self.series_name.lower() == video.series.lower():
+            if video.series and \
+               IGNORED_CHARACTERS_RE.sub('', self.series_name).lower() == \
+                IGNORED_CHARACTERS_RE.sub('', video.series).lower():
                 matches.add('series')
             # season
             if video.season and self.series_season == video.season:
@@ -77,7 +80,9 @@
         if video.imdb_id and self.movie_imdb_id == video.imdb_id:
             matches.add('imdb_id')
         # title
-        if video.title and self.movie_name.lower() == video.title.lower():
+        if video.title and \
+           IGNORED_CHARACTERS_RE.sub('', self.movie_name).lower() == \
+           IGNORED_CHARACTERS_RE.sub('', video.title).lower():
             matches.add('title')
         return matches
 
diff -Naur subliminal-0.7.4/subliminal/providers/podnapisi.py subliminal-0.7.4.patched/subliminal/providers/podnapisi.py
--- subliminal-0.7.4/subliminal/providers/podnapisi.py	2014-10-18 16:02:51.000000000 -0400
+++ subliminal-0.7.4.patched/subliminal/providers/podnapisi.py	2014-10-18 16:06:04.000000000 -0400
@@ -12,6 +12,7 @@
 import guessit
 import requests
 from . import Provider
+from . import IGNORED_CHARACTERS_RE
 from .. import __version__
 from ..exceptions import InvalidSubtitle, ProviderNotAvailable, ProviderError
 from ..subtitle import Subtitle, is_valid_subtitle, compute_guess_matches
@@ -45,7 +46,9 @@
         # episode
         if isinstance(video, Episode):
             # series
-            if video.series and self.series.lower() == video.series.lower():
+            if video.series and \
+               IGNORED_CHARACTERS_RE.sub('', self.series).lower() == \
+               IGNORED_CHARACTERS_RE.sub('', video.series).lower():
                 matches.add('series')
             # season
             if video.season and self.season == video.season:
@@ -59,7 +62,9 @@
         # movie
         elif isinstance(video, Movie):
             # title
-            if video.title and self.title.lower() == video.title.lower():
+            if video.title and \
+               IGNORED_CHARACTERS_RE.sub('', self.title).lower() == \
+               IGNORED_CHARACTERS_RE.sub('', video.title).lower():
                 matches.add('title')
             # year
             if video.year and self.year == video.year:
diff -Naur subliminal-0.7.4/subliminal/providers/tvsubtitles.py subliminal-0.7.4.patched/subliminal/providers/tvsubtitles.py
--- subliminal-0.7.4/subliminal/providers/tvsubtitles.py	2014-10-18 16:02:47.000000000 -0400
+++ subliminal-0.7.4.patched/subliminal/providers/tvsubtitles.py	2014-10-18 16:08:43.000000000 -0400
@@ -10,6 +10,7 @@
 import charade
 import requests
 from . import Provider
+from . import IGNORED_CHARACTERS_RE
 from .. import __version__
 from ..cache import region
 from ..exceptions import InvalidSubtitle, ProviderNotAvailable, ProviderError
@@ -114,7 +115,9 @@
             if not match:
                 logger.warning('Could not parse %r', link.string)
                 continue
-            if match.group('series').lower().replace('.', ' ').strip() == series.lower():
+            if IGNORED_CHARACTERS_RE.sub('', match.group('series'))\
+                    .strip().replace('.', ' ').lower() == \
+               IGNORED_CHARACTERS_RE.sub('', series).lower():
                 return int(link['href'][8:-5])
         return int(links[0]['href'][8:-5])
 
@@ -140,7 +143,7 @@
         return episode_ids
 
     def query(self, series, season, episode):
-        show_id = self.find_show_id(series.lower())
+        show_id = self.find_show_id(IGNORED_CHARACTERS_RE.sub('', series.lower()))
         if show_id is None:
             return []
         episode_ids = self.find_episode_ids(show_id, season)
