diff -Naur subliminal-0.7.4/subliminal/api.py subliminal-0.7.4.patched/subliminal/api.py
--- subliminal-0.7.4/subliminal/api.py	2014-11-11 13:49:49.000000000 -0500
+++ subliminal-0.7.4.patched/subliminal/api.py	2014-11-11 14:57:54.000000000 -0500
@@ -312,6 +312,9 @@
                 provider.terminate()
             except ProviderNotAvailable:
                 logger.warning('Provider %r is not available, unable to terminate', provider_name)
+            except socket_error as err:
+                logger.warning('Provider %r is not available, unable to terminate', provider_name)
+                logger.debug('Provider socket error: %r', str(err))
             except:
                 logger.exception('Unexpected error in provider %r', provider_name)
     return downloaded_subtitles
