diff -Naur subliminal-0.7.4.py26/subliminal/api.py subliminal-0.7.4.clean/subliminal/api.py
--- subliminal-0.7.4.py26/subliminal/api.py	2014-07-09 18:36:32.000000000 -0400
+++ subliminal-0.7.4.clean/subliminal/api.py	2014-08-21 13:29:04.000000000 -0400
@@ -233,9 +233,10 @@
                 if subtitle.provider_name in discarded_providers:
                     logger.debug('Skipping subtitle from discarded provider %r', subtitle.provider_name)
                     continue
-                if subtitle.hearing_impaired != hearing_impaired:
-                    logger.debug('Skipping subtitle: hearing impaired != %r', hearing_impaired)
-                    continue
+                if hearing_impaired is not None:
+                    if subtitle.hearing_impaired != hearing_impaired:
+                        logger.debug('Skipping subtitle: hearing impaired != %r', hearing_impaired)
+                        continue
                 if score < min_score:
                     logger.debug('Skipping subtitle: score < %d', min_score)
                     continue
