diff -Naur subliminal-0.7.4/subliminal/cli.py subliminal-0.7.4.guessit7/subliminal/cli.py
--- subliminal-0.7.4/subliminal/cli.py	2014-01-27 16:17:14.000000000 -0500
+++ subliminal-0.7.4.guessit7/subliminal/cli.py	2014-07-09 21:53:09.000000000 -0400
@@ -146,7 +146,7 @@
                          embedded_subtitles=not args.force, age=args.age)
 
     # guess videos
-    videos.extend([Video.fromguess(os.path.split(p)[1], guessit.guess_file_info(p, 'autodetect')) for p in args.paths
+    videos.extend([Video.fromguess(os.path.split(p)[1], guessit.guess_file_info(p, info=['filename'])) for p in args.paths
                    if not os.path.exists(p)])
 
     # download best subtitles
diff -Naur subliminal-0.7.4/subliminal/video.py subliminal-0.7.4.guessit7/subliminal/video.py
--- subliminal-0.7.4/subliminal/video.py	2014-01-27 16:17:14.000000000 -0500
+++ subliminal-0.7.4.guessit7/subliminal/video.py	2014-07-09 23:08:08.000000000 -0400
@@ -182,7 +182,7 @@
     """
     dirpath, filename = os.path.split(path)
     logger.info('Scanning video %r in %r', filename, dirpath)
-    video = Video.fromguess(path, guessit.guess_file_info(path, 'autodetect'))
+    video = Video.fromguess(path, guessit.guess_file_info(path, info=['filename']))
     video.size = os.path.getsize(path)
     if video.size > 10485760:
         logger.debug('Size is %d', video.size)
